### PR TITLE
Wget tests

### PR DESCRIPTION
**Pull request by rjoleary**
This also adds a few improvements:

- More complete man page
- Return 1 if the HTTP status is not 200

### CODE REVIEW

File: cmds/wget/wget.go [hunk 1]
```diff
@@ -5,37 +5,73 @@
...
 //
 // Synopsis:
-//     wget URL
+//     wget [ARGS] URL
+//
+// Description:
+//     Returns a non-zero code on failure.
+//
+// Options:
+//     -O: output filename, defaults to '-' (stdout)
```
Comment (rminnich) [inline cmds/wget/wget.go]: eek. Let's not bring  in that - notation. stdout has a name on modern operating systems :-)
You can do this:
wget_linux.go has
const stdoutName = "/dev/fd/0"

then as we get to other kernels we can work this out. But no more of these names that are not names :-)

File: cmds/wget/wget.go [hunk 2]
```diff
@@ -5,37 +5,73 @@
...
+	"flag"
+	"fmt"
 	"io"
 	"log"
 	"net/http"
 	"os"
 )
 
+var (
+	outFilename = flag.String("O", "-", "output filename, '-' for stdout")
```
Comment (rminnich) [inline cmds/wget/wget.go]: "O", stdoutName, "output filename " + stdoutName + " for stdout"

File: cmds/wget/wget.go [hunk 3]
```diff
@@ -5,37 +5,73 @@
...
-	if len(os.Args) < 2 {
-		os.Exit(1)
+	flag.Parse()
+	if flag.NArg() != 1 {
+		usage()
+	}
+
+	url := flag.Arg(0)
+	w := os.Stdout
+	if *outFilename != "-" {
```
Comment (rminnich) [inline cmds/wget/wget.go]: And then if we want we can always just open it ...

File: cmds/wget/wget.go (hunk 1 above)
Comment (rjoleary) [inline cmds/wget/wget.go]: I'd prefer not implementing a `-O` at all and just relying on the shell's redirect `wget http://example.com > out.html`.

I only implemented it so I could do a parity test with GNU's wget: `EXECPATH=wget go test`.

Thinking about it more, a better solution is to get rid of `-O` altogether, and make it so the parity test can be run like:
`EXECPATH="wget -O -" go test`.

File: cmds/wget/wget.go (hunk 2 above)
Comment (rjoleary) [inline cmds/wget/wget.go]: I'm going to remove this flag.

File: cmds/wget/wget.go (hunk 3 above)
Comment (rjoleary) [inline cmds/wget/wget.go]: I'm going to remove this flag...

File: cmds/wget/wget_test.go [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+	// Start a webserver on a free port.
+	port := getFreePort(t)
+	unusedPort := getFreePort(t)
+	h := handler{make(chan int, 1)}
+	go func() {
+		t.Fatal(http.ListenAndServe(fmt.Sprintf(":%d", port), h))
+	}()
+
+	time.Sleep(500 * time.Millisecond) // TODO: better synchronization
```
Comment (rjoleary) [inline cmds/wget/wget_test.go]: Any way of synchronizing better?
`ListenAndServe` does not seem to expose a mechanism for waiting until the server is up.

**Comment by rjoleary** [comment]
@rminnich  Please take another look!

It turns out that Travis does not support ipv6 `::1`, so I disabled that test for now.

Also, the `-O` flag is now removed. For the parity test, run `EXECPATH="wget -O -" go test`. This will pass all the tests to your system's `wget` and prepend `-O -` to the args to behave like u-root's wget.

u-root's and gnu's wget pass the same tests with one difference: u-root always returns `1` in case of error while gnu has various numbers for different failures.

File: cmds/wget/wget_test.go [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+	"time"
+
+	"github.com/u-root/u-root/shared/testutil"
+)
+
+const content = "Very simple web server"
+
+type handler struct{}
+
+func (h handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```
Comment (rminnich) [inline cmds/wget/wget_test.go]: that's nice!

File: cmds/wget/wget_test.go [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+		url:     "http://localhost:%[1]d/200",
+		stdout:  content,
+		retCode: 0,
+	}, {
+		// ipv4
+		flags:   []string{},
+		url:     "http://127.0.0.1:%[1]d/200",
+		stdout:  content,
+		retCode: 0,
+	}, /*{ TODO: travis does not support ipv6
```
Comment (rminnich) [inline cmds/wget/wget_test.go]: aaaaaaarrrrrrrrrggggggghhhhhhhhhhhhhhh

File: cmds/wget/wget_test.go (hunk 4 above)
Comment (rminnich) [inline cmds/wget/wget_test.go]: there's probably no better way than this. There are things outside of the http library control, namely how long whatever kernel you are on takes to do this.

**Pull request merged**